src/v1_0/rws_client.cpp

```diff
@@ -250,20 +250,41 @@ std::string RWSClient::generateFilePath(const FileResource& resource)
   return Services::FILESERVICE + "/" + resource.directory + "/" + resource.filename;
 }
 
+// Try finding xml tag with class="code" containing retcode.
+std::optional<int> tryParseRetcode(Poco::AutoPtr<Poco::XML::Document> const& xml)
+{
+  std::string code_str = xmlFindTextContent(xml, XMLAttribute{"class", "code"});
+  try
+  {
+    return std::stoi(code_str);
+  }
+  catch (std::invalid_argument)
+  {
+    return {};
+  }
+  catch (std::out_of_range)
+  {
+    return {};
+  }
+}
 
 POCOResult RWSClient::httpGet(const std::string& uri,
   std::set<Poco::Net::HTTPResponse::HTTPStatus> const& accepted_status)
 {
   POCOResult const result = http_client_.httpGet(uri);
 
   if (accepted_status.find(result.httpStatus()) == accepted_status.end())
-    BOOST_THROW_EXCEPTION(ProtocolError {"HTTP response status not accepted"}
+  {
+    auto exception = ProtocolError {"HTTP response status not accepted"}
       << HttpMethodErrorInfo {"GET"}
       << UriErrorInfo {uri}
       << HttpStatusErrorInfo {result.httpStatus()}
       << HttpResponseContentErrorInfo {result.content()}
-      << HttpReasonErrorInfo {result.reason()}
-    );
+      << HttpReasonErrorInfo {result.reason()};
+    if (auto retcode = tryParseRetcode(parseXml(result.content())))
```

Comment:
What if `parseXml()` throws?

Reply:
Right, I'll add a general `catch (...)` to handle any exception.

Reply:
No, not any. Just those that you would expect.

Reply:
Hmm, we are already trying to throw an exception, so I guess `tryParseRetcode` should be `noexcept` and it should catch any exception inside. 
If not, I'm not sure what kind of exceptions POCO may throw so I guess it should be `std::exception`. 
But I don't know if it would be better to throw some other exception instead of the http exception that already contains the xml content we are trying to extract the recode from.

Reply:
> If not, I'm not sure what kind of exceptions POCO may throw so I guess it should be `std::exception`.
Don't guess, read the docs ;-) :
https://docs.pocoproject.org/current/Poco.XML.XMLException.html